server/src/main/java/org/opensearch/index/mapper/MapperService.java

```diff
@@ -690,6 +710,10 @@ public boolean isFieldPartOfCompositeIndex(String field) {
         return fieldsPartOfCompositeMappings.contains(field);
     }
 
+    public boolean isCompositeIndexFieldNestedField(String field) {
```

Comment:
Is there a use-case where this method can be used separately from the `isFieldPartOfCompositeIndex`? Otherwise, can we combine both methods?